controllers/glanceapi_controller.go

```diff
@@ -1320,6 +1330,38 @@ func (r *GlanceAPIReconciler) deleteJob(
 	return ctrlResult, err
 }
 
+// getEndpointID - returns the endpointID associated to a keystone Endpoint
+func (r *GlanceAPIReconciler) getEndpointID(
+	ctx context.Context,
+	instance *glancev1.GlanceAPI,
+) (string, error) {
+	ep := &keystonev1.KeystoneEndpoint{}
+	epID := ""
+	epType := endpoint.EndpointInternal
+	// in case of split, the external API will point to its own keystone Endpoint
+	// instead of looking for a different API that might not exist or have issues
+	// we can't control here
+	if instance.Spec.APIType == glancev1.APIExternal {
+		epType = endpoint.EndpointPublic
+	}
+	err := r.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, ep)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		// Request object not found, GlanceAPI can't be executed with this config
+		r.Log.Info("EndpointID not ready")
+		instance.Status.Conditions.Set(condition.FalseCondition(
```

Comment:
-1 to this: if we have multiple APIs this is not going to work: we won't have a CR because it shouldn't be created, but we shouldn't block the process.